src/Core/Settings.h

```diff
@@ -280,6 +280,7 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(UInt64, http_max_fields, 1000000, "Maximum number of fields in HTTP header", 0) \
     M(UInt64, http_max_field_name_size, 1048576, "Maximum length of field name in HTTP header", 0) \
     M(UInt64, http_max_field_value_size, 1048576, "Maximum length of field value in HTTP header", 0) \
+    M(UInt64, http_max_chunk_size, 100_GiB, "Maximum value of a chunk size in HTTP chunked transfer encoding", 0) \
```

Comment:
@CheSema What is the reason for such a huge default? It is chunks not the whole request after all.

Reply:
I would not call it as default. It is limitation. When you exceeded it you get an error.

Reply:
It easy could be set with less value if such big values might break some logic after.

Reply:
>I would not call it as default. It is limitation. When you exceeded it you get an error.

My bad, I meant "default for max chunk size"

Reply:
Not an issue, but just a bad "recommendation" for users who will see such a big value and may think that it is a good idea to use such huge chunks.

Actually I came to this patch while I was looking at https://github.com/ClickHouse/ClickHouse/issues/55148

>Such rough limitation just enough for protection from integer overflow cases when server tries to allocate size near max<int_64>.

This is not a limitation, user can override it anyway.

Reply:
> This is not a limitation, user can override it anyway.

You are right. Maybe that setting should not exist.

Reply:
This is ok; we have a lot of settings of this type - "factory limits"

Reply:
OK, how about https://github.com/ClickHouse/ClickHouse/pull/60852